### PR TITLE
Mptrie proof api moving to pkg

### PR DESCRIPTION
**Pull request by gennadylaventman**
Moving internal `mptrie.Proof` struct to pkg `state.Proof` with all relevant methods. 

To make it available to sdk.

### CODE REVIEW

File: pkg/state/proof.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import (
+	"bytes"
+
+	"github.com/IBM-Blockchain/bcdb-server/pkg/crypto"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+	"github.com/pkg/errors"
+)
+
+var DeleteBytes = []byte{1}
```
Comment (tock-ibm) [inline pkg/state/proof.go:11]: This means "Key Deletion Marker" or "Key Deletion Tombstone" right? 
I suggest to rename to a more descriptive name...

File: pkg/state/proof.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+			return false, err
+		}
+	}
+	// Check if calculated root hash if equal to supplied (stored in block)
+	return bytes.Equal(rootHash, hashToFind), nil
+}
+
+func (p *Proof) GetPath() []*types.MPTrieProofElement {
+	return p.Path
+}
```
Comment (tock-ibm) [inline pkg/state/proof.go:70]: Why is this needed if Path is exported? 
maybe Path -> path?

**Review by tock-ibm — COMMENTED** [review]
Looks good, had just a couple of nits...

File: pkg/state/proof.go (hunk 1 above)
Comment (gennadylaventman) [reply pkg/state/proof.go:11]: Ok

File: pkg/state/proof.go (hunk 2 above)
Comment (gennadylaventman) [reply pkg/state/proof.go:70]: Let me check in sdk

**Review by tock-ibm — APPROVED** [review]
